tesseract/tesseract_planning/tesseract_motion_planners/examples/freespace_example.cpp

```diff
@@ -0,0 +1,235 @@
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <console_bridge/console.h>
+#include <opw_kinematics/opw_parameters.h>
+#include <class_loader/class_loader.hpp>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_kinematics/opw/opw_inv_kin.h>
+#include <tesseract_kinematics/core/utils.h>
+
+#include <tesseract_motion_planners/ompl/problem_generators/default_problem_generator.h>
+#include <tesseract_motion_planners/ompl/profile/ompl_default_plan_profile.h>
+#include <tesseract_motion_planners/ompl/ompl_motion_planner.h>
+
+#include <tesseract_motion_planners/trajopt/trajopt_motion_planner.h>
+#include <tesseract_motion_planners/trajopt/problem_generators/default_problem_generator.h>
+#include <tesseract_motion_planners/trajopt/profile/trajopt_default_plan_profile.h>
+#include <tesseract_motion_planners/trajopt/profile/trajopt_default_composite_profile.h>
+
+#include <tesseract_motion_planners/core/types.h>
+#include <tesseract_motion_planners/core/utils.h>
+#include <tesseract_motion_planners/interface_utils.h>
+
+#include <tesseract_visualization/visualization_loader.h>
+
+using namespace tesseract_planning;
+
+std::string locateResource(const std::string& url)
+{
+  std::string mod_url = url;
+  if (url.find("package://tesseract_support") == 0)
+  {
+    mod_url.erase(0, strlen("package://tesseract_support"));
+    size_t pos = mod_url.find('/');
+    if (pos == std::string::npos)
+    {
+      return std::string();
+    }
+
+    std::string package = mod_url.substr(0, pos);
+    mod_url.erase(0, pos);
+    std::string package_path = std::string(TESSERACT_SUPPORT_DIR);
+
+    if (package_path.empty())
+    {
+      return std::string();
+    }
+
+    mod_url = package_path + mod_url;
+  }
+
+  return mod_url;
+}
+
+int main(int /*argc*/, char** /*argv*/)
+{
+  // Setup
+  tesseract_scene_graph::ResourceLocator::Ptr locator =
+      std::make_shared<tesseract_scene_graph::SimpleResourceLocator>(locateResource);
+  auto tesseract = std::make_shared<tesseract::Tesseract>();
+  boost::filesystem::path urdf_path(std::string(TESSERACT_SUPPORT_DIR) + "/urdf/abb_irb2400.urdf");
+  boost::filesystem::path srdf_path(std::string(TESSERACT_SUPPORT_DIR) + "/urdf/abb_irb2400.srdf");
+  tesseract->init(urdf_path, srdf_path, locator);
+
+  // Dynamically load ignition visualizer if exist
+  std::string library_path = "/home/larmstrong/catkin_ws/trajopt_only_ws/devel/lib/"
+                             "libtesseract_ignition_visualization_plugin.so";
+  std::string class_name = "tesseract_ignition::TesseractIgnitionVisualization";
+  tesseract_visualization::VisualizationLoader loader(library_path, class_name);
```

Comment:
You are using a hardcoded file path from your system. 
I think it should work with just `tesseract_visualization::VisualizationLoader loader;`

Reply:
I also cannot get the visualization to load on my system, I just installed the tesseract-ignition snap, but I get a warning that it fails to load the library whenever I try and run the examples

Reply:
Version 10 is the one that should work, but it is currently in manual review and should be released soon. Once it is then should work.